src/Discord.Net/IDiscordClient.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Threading.Tasks;
+using Wumpus;
+
+namespace Discord
+{
+    public interface IDiscordClient
+    {
+        event Action Connected;
+        event Action Disconnected;
+        event Action Ready;
+
+        event Action<IChannel> ChannelCreated;
+        event Action<IChannel> ChannelDeleted;
+
+        event Action<IMessage> MessageCreated;
+        // todo: implement the rest of the events
+        // review: event classes or just Action<A,B,C> types
```

Comment:
the EventArgs pattern is probably better.

Reply:
`EventArgs` or similar please. For one, we do basically that anyway, in a wrapper layer around the client. For two, it's WAY better from a documentation standpoint. For example, `GuildMemberUpdated` today has a signature of 

```cs
public event Func<SocketGuildUser, SocketGuildUser, Task> GuildMemberUpdated;
```

If I don't already know how this event works, and want to know, which user is the "old" version and which is the "new" one, the only way to really know for sure is to go dig through the source on github and see that when it's actually invoked, it's...

```cs
await TimedInvokeAsync(_guildMemberUpdatedEvent, nameof(GuildMemberUpdated), before, user).ConfigureAwait(false);
```